scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -297,20 +297,26 @@ abstract class LocalScanner(
     }
 
     private fun createMissingArchives(scanResults: Map<Package, List<ScanResult>>, downloadDirectory: File) {
+        val missingArchives = mutableSetOf<Pair<Package, KnownProvenance>>()
+
         scanResults.forEach { (pkg, results) ->
-            val missingArchives = results.mapNotNullTo(mutableSetOf()) { result ->
-                result.provenance.takeUnless { it is KnownProvenance && archiver.hasArchive(it) }
+            results.forEach { (provenance, _, _) ->
```

Comment:
Using destructuring for data classes makes the code dependent on the order and number of properties in the data class. IMO this is an anti-pattern we should avoid.

Reply:
TBH I mostly decided to use a destructing declaration here to benefit from the smart case to `KnownProvenance` in the line below, which otherwise would have required to introduce another variable. Also in this case, reordering of the properties in `ScanResult` should lead to a compile error, so at least it does not go unnoticed (like in other cases when properties of the same type are reordered). Taking these arguments into account, would you be fine with keeping the code as-is @mnonnenmacher?